src/components/o-modal/index.js

```diff
@@ -0,0 +1,63 @@
+import PropTypes from '../../js/prop-types';
+import BaseComponentGlobal from '../../js/abstract/base-component-global';
+import defineOnce from '../../js/define-once';
+import styles from './index.scss';
+import template from './_template';
+
+class AXAModal extends BaseComponentGlobal {
+  static tagName = 'axa-modal';
+  static propTypes = {
+    open: PropTypes.bool,
+  };
+
+  init() {
+    super.init({ styles, template });
+  }
+
+  static get observedAttributes() {
+    return ['open'];
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    const axaButtons = document.querySelectorAll('axa-button[data-modal-close]');
+    if (axaButtons && axaButtons.length) {
+      axaButtons.forEach((button) => {
+        button.onclick = this.closeModal;
+      });
+    }
+
+    document.addEventListener('keyup', (event) => {
```

Comment:
please be scoped with event listeners as otherwise when listens to keyup, all the other one will. Use `this.` instead